specs/core/0_beacon-chain.md

```diff
@@ -1705,19 +1698,19 @@ def process_attestation(state: BeaconState, attestation: Attestation) -> None:
         proposer_index=get_beacon_proposer_index(state),
     )
 
-    if data.target_epoch == get_current_epoch(state):
-        ffg_data = (state.current_justified_epoch, state.current_justified_root, get_current_epoch(state))
+    if data.target.epoch == get_current_epoch(state):
+        ffg_source = state.current_justified_checkpoint
         parent_crosslink = state.current_crosslinks[data.crosslink.shard]
         state.current_epoch_attestations.append(pending_attestation)
     else:
-        ffg_data = (state.previous_justified_epoch, state.previous_justified_root, get_previous_epoch(state))
+        ffg_source = state.previous_justified_checkpoint
         parent_crosslink = state.previous_crosslinks[data.crosslink.shard]
         state.previous_epoch_attestations.append(pending_attestation)
 
-    # Check FFG data, crosslink data, and signature
-    assert ffg_data == (data.source_epoch, data.source_root, data.target_epoch)
+    # Check FFG source, crosslink data, and signature
+    assert ffg_source == data.source
```

Comment:
Do we also need to check that `data.target_epoch` matches?

Reply:
Do to the initial assert that `target_epoch in prev or cur` and the conditional statement above that filters the proper source by the attestation's target_epoch, I believe this assertion actually covers it